src/options.cpp

```diff
@@ -1035,6 +1038,38 @@ std::vector<options_manager::id_and_option> options_manager::build_soundpacks_li
     return result;
 }
 
+std::vector<std::string> options_manager::get_lang_list()
+{
+    std::vector<std::string> lang_dirs = get_directories_with( PATH_INFO::lang_file(),
+                                         PATH_INFO::langdir(), true );
+    const std::string start_str = "mo/";
+    const std::size_t start_len = start_str.length();
+    const std::string end_str = "/LC_MESSAGES";
+    std::for_each( lang_dirs.begin(), lang_dirs.end(), [&]( std::string & dir ) {
+        const std::size_t start = dir.find( start_str ) + start_len;
+        const std::size_t len = dir.rfind( end_str ) - start;
+        dir = dir.substr( start, len );
+    } );
+    return lang_dirs;
+}
+
+std::vector<options_manager::id_and_option> options_manager::get_actual_lang_options()
+{
+    std::vector<std::string> lang_list = options_manager::get_lang_list();
+    std::vector<options_manager::id_and_option> options;
+
+    lang_list.push_back( "" ); // for System language option
+    lang_list.push_back( "en" ); // for English option
+
+    std::copy_if( options_manager::lang_options.begin(),
```

Comment:
It's a little dangerous to have the initialization of one static variable depend on another like this, because of the Static Object Initialization Order Fiasco.  It will work in this case because they are both in the same translation unit and in the correct order, but if anyone rearranged the code it could break things.

If this is now the only place where `options_manager::lang_options` is used, then I suggest simply moving the hardcoded list into this function and removing that variable entirely.

If it's *not* the only place, then I suggest replcing it with a function that returns the hardcoded list, to ensure that things are created in the correct order.